lib/models/channel_model.dart

```diff
@@ -41,9 +41,8 @@ class ChannelModel extends ChangeNotifier {
     return this;
   }
 
-  bool handleNotifications(Event notification) {
-    channel!.send('run', notification.toMap());
-    return true;
+  void sendEvent(Event notification, Function(Map<dynamic, dynamic>) callback) {
```

Comment:
could you change this function to return a future rather than a callback ? 
It is probably more convenient to use

Reply:
Note: Use [Completer](https://api.dart.dev/stable/2.10.5/dart-async/Completer-class.html) to create a new future and add the receive("error") to handle the error response